src/main/java/org/spongepowered/api/event/entity/living/humanoid/player/KickPlayerEvent.java

```diff
@@ -25,11 +25,25 @@
 package org.spongepowered.api.event.entity.living.humanoid.player;
 
 import org.spongepowered.api.entity.living.player.Player;
-import org.spongepowered.api.event.message.MessageChannelEvent;
+import org.spongepowered.api.text.Text;
 
 /**
  * Fired when a {@link Player} is kicked.
  */
-public interface KickPlayerEvent extends TargetPlayerEvent, MessageChannelEvent {
+public interface KickPlayerEvent extends TargetPlayerEvent {
```

Comment:
Yeah like I said this can still be a `MessageChannelEvent`. For the default, you can create a new channel for the recipients of kick messages (this is console only in vanilla?). The main pros for having it extend `MessageChannelEvent` are as followed

* Always be able to retrieve the original reason
* Empty text != no message, you can send empty strings through chat (this is why we have `setMessageCancelled`)
* Modify the recipients of said message